src/server/game/Entities/Player/Player.cpp

```diff
@@ -6105,7 +6105,15 @@ bool Player::RewardHonor(Unit* uVictim, uint32 groupsize, int32 honor, bool awar
 void Player::SetHonorPoints(uint32 value)
 {
     if (value > sWorld->getIntConfig(CONFIG_MAX_HONOR_POINTS))
+    {
+        if (int32 copperPerPoint = sWorld->getIntConfig(CONFIG_MAX_HONOR_POINTS_MONEY_PER_POINT))
```

Comment:
use GetOption<int32>()

Reply:
Wrong. That's used when the configs are read from the text file, not from the storage when they are already loaded.
https://github.com/azerothcore/azerothcore-wotlk/pull/10565/files#diff-d88fa29cc529859e675b29c15bcddddb5fe915116df44cac561a52773107d94cR847

Reply:
According to Winfi it should be used like this https://github.com/azerothcore/azerothcore-wotlk/commit/b50984e9bfb66b496190a7b10839d1e1331dc311#diff-9dbcc2f32c3db710fb2f3843093a5aa606c9e5716599e5f540b4d1078951ef62R617

Reply:
There are two methods that can read configs, one is used to read the text file directly and the other reads the server memory.

To not be confused with the old load method that was replaced with the GetOption<type>

I don't really think it's efficient to load the text file for the config string every time the config is called. Modules do that because there's no other option, but the memory storage is probably more performance friendly and the correct way to access them during runtime.

Reply:
That said... I believe those configs you added on that other PR should've been added to the server storage (like the rest of them)

Reply:
Now I am no fan of the way the "new" configs work, that you define the default everywhere. And I do much more prefer the way it always have been done to be defined only once.

I made the suggestion only because I trust Winfi as he reworked the system

Reply:
I think there was a mistake when your PR got reviewed. It's not new, nothing changed, you can check and you'll notice every single config variable is still loaded the way it is in this PR.

You're right, it's also not good to define defaults more than once in case the same config is used multiple times.